src/routes.js

```diff
@@ -44,4 +45,24 @@ router.post('/coffeegrams', async ctx => {
   ctx.redirect(`/users/${ctx.state.currentUser.username}`);
 });
 
+router.post('/users/:username', async ctx => {
+  var user = await User.findOne({ username: ctx.params.username });
+  if (!user) {
+    throw new NotFound();
+  }
+  if (user.id !== ctx.state.currentUser.id) {
+    ctx.request.flash('error', 'You\'re not allowed to edit someone else\'s profile');
+    return ctx.redirect('back');
+  }
+  var { bio } = ctx.request.body;
+  if (isBlank(bio)) {
+    ctx.request.flash('error', 'Your bio looks empty!');
+    return ctx.redirect('back');
+  }
+  var id = user.id;
```

Comment:
Minor stylistic thing, I don't think this is improving the readability or DRYness of your code since you only reference the id once. I'd directly access `user.id` below instead of assigning to a local var. It also is strange because it's separated by another line (`clean(bio)`) from where it's actually used.